packages/extension/src/ui/utils/tokens.ts

```diff
@@ -131,19 +68,29 @@ export const fetchTokenDetails = async (
       )
       .catch(() => ""),
   ])
-  const localStorageBackup: TokenDetails = JSON.parse(
-    localStorage.getItem(`wallet:${walletAddress}:token:${address}`) || "{}",
-  )
-  const defaultBackup = defaultErc20s[address] || {}
-  const decimalsBigNumber = BigNumber.from(
-    decimals || localStorageBackup.decimals || defaultBackup.decimals || 0,
-  )
+  const decimalsBigNumber = BigNumber.from(decimals || 0)
   return {
     address,
-    name: name || localStorageBackup.name || defaultBackup.name,
-    symbol: symbol || localStorageBackup.symbol || defaultBackup.symbol,
+    name,
+    symbol,
     balance,
     networkId,
     decimals: decimalsBigNumber.isZero() ? undefined : decimalsBigNumber,
   }
 }
+
+export const fetchTokenBalance = async (
+  address: string,
+  walletAddress: string,
+  networkId: string,
+): Promise<BigNumber> => {
+  const provider = getProvider(networkId)
+  const tokenContract = new Contract(parsedErc20Abi as Abi[], address, provider)
+  const balance = await tokenContract
+    .call("balanceOf", { user: walletAddress })
+    .then((x) =>
+      BigNumber.from(uint256.uint256ToBN(x.balance as any).toString()),
+    )
+    .catch(() => BigNumber.from(0))
+  return balance
```

Comment:
Callback based programming... no

```suggestion
  try {
    const result = await tokenContract.call("balanceOf", { user: walletAddress })
    return BigNumber.from(uint256.uint256ToBN(result.balance as any).toString())
  } catch {
    return BigNumber.from(0)
  }
```